operators/src/engine/operator.rs

```diff
@@ -33,7 +33,7 @@ pub trait RasterOperator: CloneableRasterOperator + Send + Sync + std::fmt::Debu
 pub trait VectorOperator: CloneableVectorOperator + Send + Sync + std::fmt::Debug {
     fn initialize(
         self: Box<Self>,
-        context: &ExecutionContext,
+        context: &dyn ExecutionContext,
```

Comment:
maybe. But this way you only have one trait instead of n traits. And you pass the context around a lot and you would use a reference anyway. I think it's acceptable here but I can change it if you two think its worse than a generic